ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -1362,27 +1363,54 @@ void McapHandler::check_and_free_space_()
         return;
     }
 
-    if (configuration_.mcap_output_settings.file_rotation)
+    if (!configuration_.mcap_output_settings.file_rotation)
     {
-        const auto oldest_file = mcap_filenames_.front();
-        const auto oldest_file_size = std::filesystem::file_size(oldest_file);
-
-        logInfo(
-            DDSRECORDER_MCAP_HANDLER,
-            "RESOURCE_LIMITS | Removing file " << oldest_file << " on file rotation.");
+        // File rotation mode is disabled.
+        return;
+    }
 
-        const auto ret = std::filesystem::remove(oldest_file);
+    // Find the oldest file that still exists
+    std::string oldest_file;
 
-        if (!ret)
+    while (true)
```

Comment:
Isn't it possible to end up with an infinite loop?

Reply:
No, because `mcap_filenames_` is a finite vector. If `mcap_filenames_` is empty, we break out of the loop, and if it isn't, we pop its first element.

Reply:
But what if all files in the vector are deleted?